### DISCUSSION TITLE
401 Unauthorized

### DISCUSSION THREAD

**Discussion by andyweb** [fattureincloud/api#399] [Q&A] [answered]
Ciao, non riesco a capire perché mi da sempre questa risposta: 401 Unauthorized. Ho creato l'app (privata) da fatture in cloud, l'ho collegata all'azienda e generato il token. Come lancio il mio codice php dal mio server pubblico per vedere se rispondono le api, mi da sempre: 401 Unauthorized.

```
<?php
require_once(__DIR__ . '/vendor/autoload.php');


// Configure OAuth2 access token for authorization: OAuth2AuthenticationCodeFlow
$config = FattureInCloud\Configuration::getDefaultConfiguration()->setAccessToken('****************************************');


$apiInstance = new FattureInCloud\Api\ClientsApi(
    // If you want use custom http client, pass your client which implements `GuzzleHttp\ClientInterface`.
    // This is optional, `GuzzleHttp\Client` will be used as default.
    new GuzzleHttp\Client(),
    $config
);
$company_id = **********; // int | The ID of the company.



$entity = new FattureInCloud\Model\Client;
$entity
    ->setType(FattureInCloud\Model\Client::COMPANY)
    ->setName("Mario Rossi")
    ->setVatNumber("47803200154")
    ->setTaxCode("RSSMRA91M20B967Q")
    ->setAddressStreet("Via Italia, 66")
    ->setAddressPostalCode("20900")
    ->setAddressCity("Milano")
    ->setAddressProvince("MI");

// Here we put our entity in the request object
$create_client_request = new FattureInCloud\Model\CreateClientRequest;
$create_client_request->setData($entity);

// Now we are all set for the final call
// Create the client: https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/docs/Api/ClientsApi.md#createclient
try {
    $result = $apiInstance->createClient($company_id, $create_client_request);
    print_r($result);
} catch (Exception $e) {
    echo 'Exception when calling ClientsApi->createClient: ', $e->getMessage(), PHP_EOL;
}
```

**Comment by emajo** [comment, marked as answer]
Ciao @andyweb, da parte nostra il 401 viene ritornato in solo due casi:
- access token errato o scaduto
- company id non corretto

Il mio consiglio è quello di verificare la correttezza di questi dati provando a fare un test di chiamata dalla nostra [Api Reference](https://developers.fattureincloud.it/api-reference).

Grazie,
Emanuele.

**Comment by andyweb** [comment]
Grazie, un'altra domanda, è possibile creare la fattura senza creare il
cliente?

Il giorno mer 24 gen 2024 alle ore 15:41 Emanuele Saccomandi <
***@***.***> ha scritto:

> Ciao @andyweb <https://github.com/andyweb>, da parte nostra il 401 viene
> ritornato in solo due casi:
>
>    - access token errato o scaduto
>    - company id non corretto
>
> Il mio consiglio è quello di verificare la correttezza di questi dati
> provando a fare un test di chiamata dalla nostra Api Reference
> <https://developers.fattureincloud.it/api-reference>.
>
> Grazie,
> Emanuele.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fattureincloud/api/discussions/399#discussioncomment-8233535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5NHAKEHMJGOHRKQTXPAXTYQEMTBAVCNFSM6AAAAABCI2UGKGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DEMZTGUZTK>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

WIM di Andrea Spurio
62026 San Ginesio (MC)
mobile: 329 8530293
www.italiawim.it

**Reply by valmoz** [reply]
Buongiorno, può trovare la risposta alla sua domanda [qui](https://developers.fattureincloud.it/docs/guides/invoice-creation#one-step-one-the-customer).

Le suggerisco comunque di cercare sulle vecchie discussioni (anche in quelle chiuse) se è già presente una risposta al suo dubbio, ed eventualmente di aprire una nuova discussione se l'argomento della nuova domanda differisce da quella precedente.
In questo modo faciliterà la ricerca di risposte da parte degli altri utenti.

Grazie
Mauro

**Comment by andyweb** [comment]
Grazie, buon lavoro.

Il giorno mer 24 gen 2024 alle ore 16:46 Mauro Valota <
***@***.***> ha scritto:

> Buongiorno, può trovare la risposta alla sua domanda qui
> <https://developers.fattureincloud.it/docs/guides/invoice-creation#one-step-one-the-customer>
> .
>
> Le suggerisco comunque di cercare sulle vecchie discussioni (anche in
> quelle chiuse) se è già presente una risposta al suo dubbio, ed
> eventualmente di aprire una nuova discussione se l'argomento della nuova
> domanda differisce da quella precedente.
> In questo modo faciliterà la ricerca di risposte da parte degli altri
> utenti.
>
> Grazie
> Mauro
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fattureincloud/api/discussions/399#discussioncomment-8234463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5NHAPAFCYHNCYKD3VAPTDYQEUEDAVCNFSM6AAAAABCI2UGKGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DEMZUGQ3DG>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

WIM di Andrea Spurio
62026 San Ginesio (MC)
mobile: 329 8530293
www.italiawim.it
